core-bundle/src/Cron/Cron.php

```diff
@@ -120,13 +124,33 @@ public function run(string $scope): void
             $repository->unlockTable();
         }
 
+        $exception = null;
+        
         // Execute all cron jobs to be run
         foreach ($cronJobsToBeRun as $cron) {
-            if (null !== $this->logger) {
-                $this->logger->debug(sprintf('Executing cron job "%s"', $cron->getName()));
+            try {
+                if (null !== $this->logger) {
+                    $this->logger->debug(sprintf('Executing cron job "%s"', $cron->getName()));
+                }
+
+                $cron($scope);
+            } catch (CronExecutionSkippedException $e) {
+                // Restore previous run date in case cronjob skips itself
+                /** @var CronJobEntity $lastRunEntity */
```

Comment:
Should we also debug log here?

Reply:
I thought about that but decided not to since it's the cronjob itself that decides to skip, not the cron service and thus if logging is needed, they would add it themselves there, like

```php
$this->logger->debug(sprintf('Skipping %s because it is not executed in the CLI scope.', __METHOD__));
```